### DISCUSSION TITLE
how to read a local json file, and return a value based on the given key?

### DISCUSSION THREAD

**Discussion by tyehuoPHD** [SilentVoid13/Templater#1075] [Help] [answered]
# how to read a local json file, and return a value based on the given key?

can anybody tell me how to properly fetch a json file?

# I have found the solution. but can i use relative path?
```js
function fetchData(key) {
  const data = require("D:/BaiduSyncdisk/ObsidianFolder/test/Templates/Scripts/dict_stock_code.json");
  return data[key];
}
module.exports = fetchData;
```

# This works smoothly. But after i replace the absolute path to relative path `./dict_stock_code.json`. it failed.
```js
function fetchData(key) {
  const data = require("./dict_stock_code.json");
  return data[key];
}
module.exports = fetchData;
```
**error:**
```log
plugin:templater-obsidian:61 Templater Error: Template parsing error, aborting. 
 Cannot find module './dict_stock_code.json'
Require stack:
- electron/js2c/renderer_init
```

## I use `process.cwd()` to find the current working dir
```js
function fetchData(key) {
  const path = require("path");
  const jsonPath = path.join(currentDirPath, "dict_stock_code.json");

  return jsonPath;
}
module.exports = fetchData;
```
it returns the installation directory of my obsidian. what should i do to use the relative path of json file ?

**Comment by Zachatoo** [comment, marked as answer]
You can use `app.vault.adapter.read` to read the contents of a text-based file relative to the root of your vault (not relative to where you called the script from).

```js
async function fetchData(app, key) {
  const data = await app.vault.adapter.read("dict_stock_code.json");
  return data[key];
}
```

You will need to pass in `app` to your user function, and await it.

```js
<% await fetchData(app, "this is a key") %>
```

More details around APIs available to you regarding processing files [here](https://github.com/obsidianmd/obsidian-api/blob/master/obsidian.d.ts#LL732C5-L732C51).

**Reply by tyehuoPHD** [reply]
thanks man ! thank you very much.
